support/src/main/java/org/jenkinsci/plugins/workflow/support/actions/WorkspaceActionImpl.java

```diff
@@ -52,9 +52,9 @@
 
     public WorkspaceActionImpl(FilePath workspace, FlowNode parent) {
         // TODO see FilePathPickle:
-        node = FilePathPickle.Listener.channelNames.get(workspace.getChannel());
+        node = FilePathPickle.Listener.getChannelName(workspace.getChannel());
         if (node == null) {
-            throw new IllegalStateException("no known slave for " + workspace + " among " + FilePathPickle.Listener.channelNames.values());
+            throw new IllegalStateException("no known slave for " + workspace + " among " + FilePathPickle.Listener.getChannelNames());
```

Comment:
You can just make `getChannelName` be `@Nonnull` and throw `IllegalStateException`; then the callers are simpler, and you do not need `getChannelNames` since that is only used for the exception message.